components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -193,8 +200,18 @@ std::unique_ptr<RuleIterator> BravePrefProvider::GetRuleIterator(
 void BravePrefProvider::UpdateCookieRules(ContentSettingsType content_type,
                                           bool incognito) {
   auto& rules = cookie_rules_[incognito];
+  auto old_rules = std::move(brave_cookie_rules_[incognito]);
+
   rules.clear();
 
+  if (prefs_->GetBoolean(kGoogleLoginControlType)) {
```

Comment:
I desperately desire a comment here, describing the whole situation. So an occasional stranger could have a little chance to understand why and what happens here :)